### PR TITLE
Validate non-negative (local_)early_items as well as [0, 0]

### PR DESCRIPTION
**Pull request by bbernardoni**
## What is this fixing or adding?

Validate that `early_items` and `local_early_items` counts are non-negative by raising a ValueError. Also validate that both are not zero for any player, item combination (and ignore in that case).

Previously, a count of `[0, 0]` was decremented to `[-1, 0]` and then compared to `[0, 0]`. Since they are unequal, the code would proceed to place all of that item in the itempool.

## How was this tested?

Unit tests passed.

### CODE REVIEW

File: Fill.py [hunk 1]
```diff
@@ -408,8 +408,13 @@ def distribute_early_items(multiworld: MultiWorld,
     for player in multiworld.player_ids:
         items = itertools.chain(multiworld.early_items[player], multiworld.local_early_items[player])
         for item in items:
-            early_items_count[item, player] = [multiworld.early_items[player].get(item, 0),
-                                               multiworld.local_early_items[player].get(item, 0)]
+            item_count = [multiworld.early_items[player].get(item, 0), multiworld.local_early_items[player].get(item, 0)]
```
Comment (duckboycool) [inline Fill.py:411]: Maybe this should be named `early_counts` or something instead? The formatting's already a bit weird here but that makes it sound like it's meant to be the number of a certain item present or something.

**Review by duckboycool — APPROVED** [review]
Tested setting either value to 0 to confirm current behavior and that it now acts the same as not being there at as expected. Also tested the negative value handling and that setting them to a normal value still works as expected.

I'm a little unsure if erroring on negative values is the best thing to do since someone for instance might set it to some expression `a - b` and if `b` is larger it should maybe still act like 0. But if that is somebody's desired behavior then currently they'd have a bug, so I'll say it's better to be loud about it now and then maybe loosen if the need arises.
